src/commands/commit.ts

```diff
@@ -24,11 +28,35 @@ export async function _commit(parsedSettings: ParsedSettings): Promise<void> {
   if (Number.isNaN(newMigrationNumber)) {
     throw new Error("Could not determine next migration number");
   }
-  const newMigrationFilename =
-    String(newMigrationNumber).padStart(6, "0") + ".sql";
-  const currentMigrationPath = getCurrentMigrationPath(parsedSettings);
-  const body = await fsp.readFile(currentMigrationPath, "utf8");
-  const minifiedBody = pgMinify(body);
+
+  // See if we have a message arg
+  const messageIndex =
+    process.argv.findIndex(arg => arg === "--message" || arg === "-m") + 1;
+
+  // If we do, fetch, and replace any whitespace with '_'
+  const messageFromCommandArgs = messageIndex && process.argv[messageIndex];
+  const messageFileContentsMatch = /--! Title:(.*)/.exec(body);
```

Comment:
This needs a "beginning of line" constraint; and also it should be careful that it doesn't match within a string within SQL. My advice would be to only look at the initial lines in the commit that start `--`, when they end stop processing.

We also have to be careful that `--! Title` doesn't conflict with the no-transaction comment which has to come first (at least currently).

We should also be consistent in terminology, using `title` here but `message` elsewhere is perhaps confusing. Can we use `Message` here?

Is this the sanitised message, or unsanitised? If unsanitised, is there any option for newlines/other special characters? Should there be?